### DISCUSSION TITLE
How to let packer install all the missing plugins automaitcally?

### DISCUSSION THREAD

**Discussion by y9c** [wbthomason/packer.nvim#224] [Q&A]
After copying config files into a new machine, It is possible to install packer.nvim automatically if it is not exist. But how to install the plugins right after packer.nvim?

**Comment by gegoune** [comment]
https://github.com/wbthomason/packer.nvim#bootstrapping talks about bootstrapping.

**Reply by y9c** [reply]
Hi cloggier, the bootstrapping section shown method for installing packer.nvim automatically, but this is not what I mean. I would like to install the plugins automatically after packer.nvim was installed.

**Reply by gegoune** [reply]
You can then add `execute 'PackerInstall'` after `packed` line. That should work.

**Reply by y9c** [reply]
Thank your @cloggier, I have tried this way before. But since packer.nvim is not sourced, no `PackerInstall` command can be found, unless I exit vim and open vim again manually.

**Reply by wbthomason** [reply]
You can use `packadd packer.nvim` to ensure that `packer` is sourced. It's probably also best to run `PackerSync` rather than `PackerInstall` to ensure that your compiled file gets correctly constructed.

**Reply by wbthomason** [reply]
To clarify: you'll want the `execute PackerSync` command **after** your call to `packer.startup` (assuming you're using the startup method), as that's what creates those commands.

**Reply by y9c** [reply]
Hi @wbthomason , do you mean add a extract line like this?
![image](https://user-images.githubusercontent.com/5415510/108529246-3f61a680-730f-11eb-9368-b508c5460871.png)

I have tried this before. It will cause a error, which is 

![image](https://user-images.githubusercontent.com/5415510/108529435-6f10ae80-730f-11eb-8f5d-c61915a5d3e6.png)

**Comment by y9c** [comment]
BTW, I mean Packer command not work after a fresh installation.
![image](https://user-images.githubusercontent.com/5415510/108529691-b139f000-730f-11eb-8396-bb545fdb6f55.png)

**Reply by wbthomason** [reply]
Right - that's why I was saying that you need the `execute PackerSync` *after* you use `packer.startup` or similar. Otherwise, the commands will not exist.
e.g.
```lua
if not packer_exists then
...
end
require('packer').startup(....)
if not packer_exists then execute 'PackerSync' end
```

**Reply by y9c** [reply]
Thank you @wbthomason . Seem that it is a little difficult for me. I wrap packer conf into a lua script and return `packer.startup(...)` in the last line of the script. I don't known how to run packersync after the return.

**Reply by y9c** [reply]
One of the attempts,

https://github.com/yech1990/myneovim.lua/tree/b0b9b3f067d8e4ea8f2e7c2a86d28596f3c037fb

**Reply by rafamadriz** [reply]
Did you find a solution ? I have the same problem.

**Reply by HicaroD** [reply]
Hey, I know this question is old, but I was having the same problem and I solved it doing the following steps:

- Remove your `plugin` folder from your `nvim` root folder.
- Reinstall Packer
  Packer is installed at that location `~/.local/share/nvim/site/pack/packer/start/packer.nvim`, just remove it and install it again.
- Now, open the lua file where Packer is being used to install plugins and run the editor command `:so %`

Now, you might be able to run commands like `PackerSync` for installing your packages.

That's how I solved it, hope you guys can solve it too.
